service/controller/v24/cluster_resource_set.go

```diff
@@ -570,6 +576,7 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*controller.Resourc
 				return nil, microerror.Mask(err)
 			}
 			c := config.HostAWSConfig
+			c.Region = key.Region(cr)
```

Comment:
Create AWSClient towards tenant cluster region instead of control plane one.

Reply:
That looks a bit dangerous. What are the implications for the generated clients? There is for instance the `accountid` resource which uses the tenant clients to gather the account ID. Does the region change here influence that somehow? When only the region changes the account ID should stay the same, right?

Reply:
It is also weird that we create a client with the `HostAWSConfig` but the tenant region.

Reply:
I don't fully understand what would be dangerous here? The thing here is that when AWS client is created for each tenant cluster, it's created for corresponding AWS region. This is like that because the client is then locked into this given region and all operations performed with it are done against this region.

As I understand this, this uses `HostAWSConfig` because the account that client is using is the one control plane is running on. Below this, a tenant specific `RoleARN` is set which I understand as being the BYOC role that works against some other AWS account via a role.

Also regarding the account id, the region doesn't affect this in any way. AWS accounts are global and so are some specific services as well (such as IAM, Route53, S3).

What kind of risks do you see here or do you have specific problem in mind that this could cuase?

Reply:
Sounds like this is "fine". The region does not affect the account. That makes sense. I am not sure about the role assumption magic but that did not change. All good.